jme3-core/src/main/java/com/jme3/effect/ParticleEmitter.java

```diff
@@ -1117,6 +1117,68 @@ private void updateParticleState(float tpf) {
 
         vars.release();
     }
+    
+    private ColorRGBA startColorToFadeTo;
+    private ColorRGBA initialStartColorToFadeFrom;
+    private float totalStartColorFadeDuration;
+    private float startColorFadeDuration;
+    
+    private ColorRGBA endColorToFadeTo;
+    private ColorRGBA initialEndColorToFadeFrom;
+    private float totalEndColorFadeDuration;
+    private float endColorFadeDuration;    
```

Comment:
startColorToFadeTo, initialStartColorToFadeFrom, endColorToFadeTo and initialEndColorToFadeFrom should be initialized to new ColorRGBA() and then the setFadeXXXX methods should call .set(inputColor) on them, it is generally safer to do it this way since it eliminates possible issues caused by misuses of shared objects.